src/test/java/api/loans/InHouseUseCheckInTest.java

```diff
@@ -0,0 +1,169 @@
+import static api.support.http.Offset.noOffset;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+import java.util.UUID;
+
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.CheckInByBarcodeResponse;
+import api.support.MultipleJsonRecords;
+import api.support.builders.RequestBuilder;
+
+public class InHouseUseCheckInTest extends APITests {
+
+  @Test
+  public void isInHouseUseWhenItemIsAvailableAndCheckInServingHomeLocation() {
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      builder -> builder.withPrimaryServicePoint(checkInServicePointId));
+
+    final IndividualResource nod = itemsFixture.basedUponNod(
+      item -> item.withTemporaryLocation(homeLocation.getId()));
+
+    final CheckInByBarcodeResponse checkInResponse = loansFixture
+      .checkInByBarcode(nod, checkInServicePointId);
+
+    assertThat(checkInResponse.getJson().containsKey("loan"), is(false));
+    assertThat(checkInResponse.getJson().containsKey("item"), is(true));
+    assertThat(checkInResponse.getInHouseUse(), is(true));
+  }
+
+  @Test
+  public void isInHouseUseWhenItemHasClosedRequests() {
+    final UUID checkInServicePointId = servicePointsFixture.cd1().getId();
+
+    final IndividualResource homeLocation = locationsFixture.basedUponExampleLocation(
+      builder -> builder.withPrimaryServicePoint(checkInServicePointId));
+
+    final IndividualResource nod = itemsFixture.basedUponNod(
+      item -> item.withTemporaryLocation(homeLocation.getId()));
+
+    requestsFixture.place(new RequestBuilder()
+      .page()
+      .forItem(nod)
+      .withPickupServicePointId(checkInServicePointId)
+      .by(usersFixture.james()));
+
+    IndividualResource recallRequest = requestsFixture.place(new RequestBuilder()
+      .recall()
+      .forItem(nod)
+      .withPickupServicePointId(checkInServicePointId)
+      .by(usersFixture.charlotte()));
+
+    requestsFixture.cancelRequest(recallRequest);
+
+    // Fulfill the page request
+    loansFixture.checkOutByBarcode(nod, usersFixture.james());
+    loansFixture.checkInByBarcode(nod, checkInServicePointId);
+
+    final CheckInByBarcodeResponse checkInResponse = loansFixture
+      .checkInByBarcode(nod, checkInServicePointId);
+
+    assertThat(checkInResponse.getJson().containsKey("loan"), is(false));
+    assertThat(checkInResponse.getJson().containsKey("item"), is(true));
+    assertThat(checkInResponse.getInHouseUse(), is(true));
+
+    MultipleJsonRecords requestsForItem = requestsFixture.getRequests(
+      queryFromTemplate("itemId=%s", nod.getId()), noLimit(), noOffset());
+    assertThat(requestsForItem.totalRecords(), is(2));
+  }
+
+  @Test
+  public void isInHouseUseWhenCheckedInToServicePointServingHomeLocation() {
```

Comment:
How is this different to the first test (`isInHouseUseWhenItemIsAvailableAndCheckInServingHomeLocation`)? 

Is one demonstrating check in at the primary service point and the other demonstrating check in at another service point that serves the home location?

Reply:
> Is one demonstrating check in at the primary service point and the other demonstrating check in at another service point that serves the home location?

Yes the `isInHouseUseWhenItemIsAvailableAndCheckInServingHomeLocation` demonstrates that item is available + check-in service point is primary service point of location + request queue is empty.

`isInHouseUseWhenCheckedInToServicePointServingHomeLocation` demonstrates that item is available + check-in SP is not primary SP of location but it present in the `servicePointIds` array for location + no requests in the queue.

Reply:
I suggest these tests be renamed to better represent these two situations
* When the check in service point is the primary service point for the home location
* When the check in service point is another service point for the home location

Does that make sense?